### PR TITLE
Package AaC Core Spec DSL in the Distribution

### LINKED ISSUE #109: I want the AaC Core Spec DSL packaged in the executable.
As a user of AaC, I want AaC DSL files that describe distribution executables packaged in the executable, so that when I install the `aac` package it will contain the `AaC.yaml` file it needs in order to execute commands like `aac aac-core-spec`.

Example code for updating util.py to read in a package resource.
```python
    # get the AaC.yaml spec for architecture modeling
    with path(f"{__package__}.model", "AaC.yaml") as aac_model_file:
        AAC_MODEL = parser.parse_file(aac_model_file, False)
```

AC:
- [x] "AaC.yml" is moved to a directory module in the AaC project 
  - [x] The aac spec directory contains the AaC Core Spec file
- [x] The AaC setup.py is set to include the yaml file. 
  - [x] setup.py:ssetup() includes this line `package_data={'aac.model': ['AaC.yaml']}`
- [x] Update the core spec module to use `importlib.resources.path` to load the AaC.yaml file.
- [x] The pipeline exercises the distribution wheel
   - [x] `validate` works
   - [x] `aac-core-spec` works

### PR DESCRIPTION
**Pull request by Coffee2Bits**
Closes #109 

* Properly configured setup.py
  * Include data files (yaml, jinja)
  * Fixed packaging to work for editable and non-editable package installation
* Added a second pipeline job `test`
  * Installs the distribution wheel
  * Checks that the distribution works as expected
 * Removed bad dependencies that broke the gitpod env (Earlier mistake on my part)

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -63,8 +54,44 @@ jobs:
...
-      #   uses: pypa/gh-action-pypi-publish@master
-      #   with:
-      #     password: ${{ secrets.PYPI_API_TOKEN }}
+      - name: Upload Python Wheel
+        uses: actions/upload-artifact@v2
+        with:
+          name: aac_wheel
+          path: dist/
+
+  test:
```
Comment (cvchaparro) [inline .github/workflows/python-app.yml:63]: Could we call this smoke test, functional test, acceptance test, or something to make it more explicit what's going on. If we just call it test, then I'd think we should move (at least) the unit test stage to this job.

File: setup.py [hunk 2]
```diff
@@ -59,16 +57,21 @@
 setup(
     name="aac",
     version="0.0.4",
+    description=(
+        "A distinctly different take on Model-Based System Engineering (MBSE) that allows a system modeller to define a system in simple yaml. "
+    ),
+    license="MIT",
```
Comment (cvchaparro) [inline setup.py:63]: Did we for sure decide to license it with the MIT license, then?

**Review by cvchaparro — CHANGES_REQUESTED** [review]

File: .github/workflows/python-app.yml (hunk 1 above)
Comment (Coffee2Bits) [reply .github/workflows/python-app.yml:63]: Good point, it is nondescript

Comment (Coffee2Bits) [reply .github/workflows/python-app.yml:63]:
```suggestion
  Wheel Package Tests:
```

File: setup.py (hunk 2 above)
Comment (Coffee2Bits) [reply setup.py:63]: It's already the Jellyfish's license. I can remove the license for now.
https://github.com/NorthropGrumman/jellyfish/blob/master/LICENSE

Comment (Coffee2Bits) [reply setup.py:63]:
```suggestion
    license="",
```

File: .github/workflows/python-app.yml [hunk 3]
```diff
@@ -12,7 +12,6 @@ on:
 jobs:
   build:
```
Comment (Coffee2Bits) [inline .github/workflows/python-app.yml:13]:
```suggestion
  Build AaC Package and Docs:
```

File: .github/workflows/python-app.yml [hunk 4]
```diff
@@ -10,9 +10,8 @@ on:
     branches: [main]
 
 jobs:
-  build:
+  Build AaC Package and Docs:
```
Comment (cvchaparro) [inline .github/workflows/python-app.yml:13]: Looks like GitHub doesn't like spaces in the name:

```suggestion
  Build-AaC-Package-and-Docs:
```

File: .github/workflows/python-app.yml [hunk 5]
```diff
@@ -63,8 +54,44 @@ jobs:
...
-      #   uses: pypa/gh-action-pypi-publish@master
-      #   with:
-      #     password: ${{ secrets.PYPI_API_TOKEN }}
+      - name: Upload Python Wheel
+        uses: actions/upload-artifact@v2
+        with:
+          name: aac_wheel
+          path: dist/
+
+  Package Functional Tests:
```
Comment (cvchaparro) [inline .github/workflows/python-app.yml:63]: See above:

```suggestion
  Package-Functional-Tests:
```

File: .github/workflows/python-app.yml [hunk 6]
```diff
@@ -63,8 +54,44 @@ jobs:
...
-      #     password: ${{ secrets.PYPI_API_TOKEN }}
+      - name: Upload Python Wheel
+        uses: actions/upload-artifact@v2
+        with:
+          name: aac_wheel
+          path: dist/
+
+  Package-Functional-Tests:
+    runs-on: ubuntu-latest
+    needs: build
```
Comment (cvchaparro) [inline .github/workflows/python-app.yml:65]:
```suggestion
    needs: Build-AaC-Package-and-Docs
```

**Review by cvchaparro — APPROVED** [review]

**Pull request merged**